docs/nomt_specification.md

```diff
@@ -0,0 +1,176 @@
+
+When a leaf is found, two possibilities arise based on the shared bits between the extracted key (let's call it 'key1') and the key we want to insert (let's call it 'key2') from the stopped key_pos during traversal. If 'get' method returns a leaf for 'key1' and 'key2', there is a shared prefix. 'Key1' is already in the tree, so a section of the shared prefix with 'key2' is also in the tree (internal nodes representing a portion of that prefix).
+
+For example, with keys of 10 bits: 
+
+|      | x      | n  | l  |
+|:-----|:-------|:---|:---|
+| key1 | 001011 | 11 | 01 |
+| key2 | 001011 | 11 | 11 |
+    
+The keys are divided into three sections - 'x' represents bits already in the tree as internal nodes, 'n' are shared bits that are not in tree, and 'l' are all the bits after a single bit difference.
+
+This division is necessary as, for each 'n' bits, an internal node with an empty child must be created, and one internal node with two leafs for all `l` bits, implicitly showing the difference in 'l' bits.
+
+A special scenario occurs when there are no 'n' bits, and only one new internal node is created.
+
+![nomt put](./images/nomt_put.png)
+
+#### Remove 
+
+To start the removal process, we first use a `get` method. If the retrieved node is empty, no action is required. If the node is a leaf, there are two scenarios:
+
+1. If the sibling of the leaf to remove is an internal node, we simply update the leaf to be empty.
+
+2. If the sibling of the leaf is another leaf (referred to as `sib_leaf`), the following steps are taken:
+   - Invalidate the leaf (which contains the value to be removed) and `sib_leaf` making them empty nodes.
+   - Traverse all ancestors, invalidating them if their sibling is an empty node.
+   - Once an ancestor with a valid sibling is found, replace it with `sib_leaf`.
+   
+##  How is this stored in memory?
+
+### Pages
+
+Nodes will be split into rootless sub-binary trees of depth `d`, so each group will contain `2^(d+1) - 2` nodes.
+
+![nomt pages](./images/nomt_pages.jpg)
+
+We will refer to each group as a *Page*. These pages will construct a **page base-(2^d) tree**, where each Page will have `2^d` child pages below it.
+
+### Node Key and Sextet
+
+A node key represents a `path` in the trie, which we will call **KeyPath**. The KeyPath is composed of 256 bits, where each bit defines a fork in the node's tree. However, each page will contain a rootless sub-binary tree, so the KeyPath can be split into multiple segments that refer to the path in the Page tree. These segments are called **Sextet**.
+
+Each sextet defines which page table to go to next, starting from the root page.
+
+If d is a divisor of the key length, then there are `256/d` Sextets, each 'd'-bits long; otherwise, Sextets are variable length. The first `⎣256/d⎦` Sextets are 'd'-bits long, while the last one is `256 - (⎣256/d⎦ * d)` bits long. (TODO: Does this make sense?)
```

Comment:
This does not make sense to me. The name sextet implies there are 6 bits in each sextet. Only full sextets provide information about the location in the page tree.

Besides, the name sextet is probably not the best suited for this explanation, which is at a level of generality of arbitrary page sizes.

Reply:
I will refactor this part and probably update also the content due to the rethinking of conversion between node_ids and page_ids works!

yes, sextet does not make any sense, I used that word as it would be an arbitrary sequence of d bits. What do you think about 'd_tets'?